airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/product_stream_slicer.py

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import itertools
+from collections import ChainMap
+from typing import Any, Iterable, List, Mapping
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.declarative.stream_slicers.stream_slicer import StreamSlicer
+
+
+class ProductStreamSlicer(StreamSlicer):
```

Comment:
nit: this sounds like "product-stream" i.e a stream called "product. Maybe call it `CompositeSlicer` or `CartesianProductSlicer`?